src/main/java/com/salesforce/dynamodbv2/mt/util/CompositeStrings.java

```diff
@@ -64,7 +64,7 @@ public String join(Iterable<String> values) {
      * @return Sequence of keys contained in the given composite key
      */
     public Iterator<String> split(String composite) {
-        return new Iterator<String>() {
+        return new Iterator<>() {
```

Comment:
`Iterator<>` sufficient as of jdk10.

Reply:
What did 10 add that prevented this from working in 1.7?

Reply:
`... new Iterator<> ...` was not supported in Java 1.6 or earlier ... `... new Iterator<String> ...` was required.